### ISSUE TITLE
Calculation of average training loss stops while training - nan

### ISSUE THREAD

**Issue by KatieBelli** [ZFTurbo/Music-Source-Separation-Training#128]
I'm on epoch 24, dataset type 4, 210 folders with wav files. I didn't change anything about the wav files.

After training the first few epochs I disabled some augmentations in the config file and increased the number of steps from 2100 to 5000. Then I resumed the training successfully.

This nan-error happens suddenly while training epoch 24. Before it reaches step 3000.

I removed the pycache folder, the metadata-pkl file, turned my pc on and off, cleaned the registry, removed pip cache and then resumed the training from the checkpoint and with the most recent epoch number in train.py but the error remains.

```
audio:
  chunk_size: 130560
  dim_f: 1024
  dim_t: 256
  hop_length: 512
  n_fft: 2048
  num_channels: 2
  sample_rate: 44100
  min_mean_abs: 0.000

model:
  act: gelu
  bottleneck_factor: 4
  growth: 128
  norm: InstanceNorm
  num_blocks_per_scale: 2
  num_channels: 128
  num_scales: 5
  num_subbands: 4
  scale:
  - 2
  - 2

training:
  batch_size: 2
  gradient_accumulation_steps: 3
  grad_clip: 0
  instruments:
  ....
  lr: 1.0
  patience: 4
  reduce_factor: 0.95
  target_instrument: null
  num_epochs: 500
  num_steps: 5000
  q: 0.95
  coarse_loss_clip: true
  ema_momentum: 0.999
  optimizer: prodigy
  other_fix: false # it's needed for checking on multisong dataset if other is actually instrumental
  use_amp: true # enable or disable usage of mixed precision (float16) - usually it must be true

augmentations:
  enable: true # enable or disable all augmentations (to fast disable if needed)
  
  all:
    channel_shuffle: 0.5 # Set 0 or lower to disable
    random_inverse: 0.01 # inverse track (better lower probability)
    random_polarity: 0.5 # polarity change (multiply waveform to -1)

inference:
  batch_size: 1
  dim_t: 256
  num_overlap: 8
```

**Comment by ZFTurbo** [comment]
You need to restart from checkpoint which was saved before first nan appeared. If you will still have nans then set use_amp: false for some epochs and reduce batch_size 2 times.

**Comment by KatieBelli** [comment]
> You need to restart from checkpoint which was saved before first nan appeared. If you will still have nans then set use_amp: false for some epochs and reduce batch_size 2 times.

It helped to resume from an earlier checkpoint, I took the checkpoint from the saved epoch 22.

I'll watch if there is something weird happening after step 3000 with the value of the average training loss. 

Is there a reason why this can happen so suddenly?

**Comment by KatieBelli** [comment]
> You need to restart from checkpoint which was saved before first nan appeared. If you will still have nans then set use_amp: false for some epochs and reduce batch_size 2 times.

After step 3000 (I train with 5000) the average loss barely changes, it's pending (now comes an example) between 0.0293 and 0.0296 but the average loss of 0.0294 is displayed for many steps. 

The average loss doesn't change much, even though before step 3000 the average loss number was changing more frequently.

Like, shouldn't the average loss number change frequently all the time while training an epoch? 

Should I just ignore this as long the "nan"-issue is gone? In the end, the calculation of the SDR is more important, right?

**Comment by KatieBelli** [comment]
Starting the training from epoch 0 and clearing all previous pycache and metadata helped. 

I guess the issue was that the model couldn't handle the changes (maybe some kind of compability issues) in the config file before resuming the training.
